tests/ut/ir/parser/test_dist_tensor_comm.py

```
@@ -0,0 +1,184 @@
+
+    walk_stmt(func.body)
+    assert found_rank == 2, f"expected 2 pld.comm_ctx.rank calls, found {found_rank}"
+
+
+# ---------------------------------------------------------------------------
+# Negative
+# ---------------------------------------------------------------------------
+
+
+def test_plain_tensor_comm_attribute_rejected():
+    with pytest.raises(ParserError, match="DistributedTensor|attribute"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)